Cannot build since #441

Since PR #441 was merged into the master, I cannot build any more. I am getting the following errors:

```
/Users/law/repos/hyrise/src/lib/optimizer/abstract_syntax_tree/abstract_ast_node.hpp:281:51: error:
      implicit instantiation of undefined template
      'std::__1::array<std::__1::shared_ptr<opossum::AbstractASTNode>, 2>'
  std::array<std::shared_ptr<AbstractASTNode>, 2> _children;
                                                  ^
/usr/local/Cellar/llvm/5.0.0/include/c++/v1/__tuple:223:64: note: template is declared
      here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
In file included from /Users/law/repos/hyrise/src/lib/optimizer/abstract_syntax_tree/abstract_non_optimizable_ast_node.cpp:1:
In file included from /Users/law/repos/hyrise/src/lib/optimizer/abstract_syntax_tree/abstract_non_optimizable_ast_node.hpp:3:
/Users/law/repos/hyrise/src/lib/optimizer/abstract_syntax_tree/abstract_ast_node.hpp:281:51: error:
      implicit instantiation of undefined template
      'std::__1::array<std::__1::shared_ptr<opossum::AbstractASTNode>, 2>'
  std::array<std::shared_ptr<AbstractASTNode>, 2> _children;
                                                  ^
/usr/local/Cellar/llvm/5.0.0/include/c++/v1/__tuple:223:64: note: template is declared
      here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
/Users/law/repos/hyrise/src/lib/optimizer/abstract_syntax_tree/abstract_ast_node.cpp:100:43: error:
      no matching member function for call to 'shared_from_this'
    current_child->_remove_parent_pointer(shared_from_this());
                                          ^~~~~~~~~~~~~~~~
/usr/local/Cellar/llvm/5.0.0/include/c++/v1/memory:5237:21: note: candidate function not
      viable: no known conversion from 'opossum::AbstractASTNode' to
      'std::__1::enable_shared_from_this<opossum::AbstractASTNode>' for object argument
    shared_ptr<_Tp> shared_from_this()
                    ^
/usr/local/Cellar/llvm/5.0.0/include/c++/v1/memory:5240:27: note: candidate function not
      viable: no known conversion from 'opossum::AbstractASTNode' to 'const
      std::__1::enable_shared_from_this<opossum::AbstractASTNode>' for object argument
    shared_ptr<_Tp const> shared_from_this() const
                          ^
/Users/law/repos/hyrise/src/lib/optimizer/abstract_syntax_tree/abstract_ast_node.cpp:108:40: error:
      no matching member function for call to 'shared_from_this'
    current_child->_add_parent_pointer(shared_from_this());
                                       ^~~~~~~~~~~~~~~~
/usr/local/Cellar/llvm/5.0.0/include/c++/v1/memory:5237:21: note: candidate function not
      viable: no known conversion from 'opossum::AbstractASTNode' to
      'std::__1::enable_shared_from_this<opossum::AbstractASTNode>' for object argument
    shared_ptr<_Tp> shared_from_this()
                    ^
/usr/local/Cellar/llvm/5.0.0/include/c++/v1/memory:5240:27: note: candidate function not
      viable: no known conversion from 'opossum::AbstractASTNode' to 'const
      std::__1::enable_shared_from_this<opossum::AbstractASTNode>' for object argument
    shared_ptr<_Tp const> shared_from_this() const
                          ^
1 error generated.
make[3]: *** [src/lib/CMakeFiles/hyrise.dir/optimizer/abstract_syntax_tree/abstract_non_optimizable_ast_node.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
3 errors generated.
make[3]: *** [src/lib/CMakeFiles/hyrise.dir/optimizer/abstract_syntax_tree/abstract_ast_node.cpp.o] Error 1
make[2]: *** [src/lib/CMakeFiles/hyrise.dir/all] Error 2
make[1]: *** [src/bin/CMakeFiles/hyrisePlayground.dir/rule] Error 2
make: *** [hyrisePlayground] Error 2
```

Does anyone know what is causing this?

--- a/src/lib/optimizer/abstract_syntax_tree/abstract_ast_node.hpp
+++ b/src/lib/optimizer/abstract_syntax_tree/abstract_ast_node.hpp
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <array>
 #include <iostream>
 #include <memory>
 #include <optional>

